ProfessorAssessor/View Controllers/LoginViewController.m

```diff
@@ -18,19 +23,77 @@ - (IBAction)login:(UIButton *)sender {
     NSString *password = self.password.text;
 
     if ([self validCredentials]) {
-        [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser *user, NSError *error) {
+        [PFUser logInWithUsernameInBackground:username
+                                     password:password
+                                        block:^(PFUser *user, NSError *error) {
             if (error == nil) {
-                UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
-                HomeViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"HomeViewController"];
-
-                SceneDelegate *sceneDelegate = (SceneDelegate *)self.view.window.windowScene.delegate;
-
-                [sceneDelegate changeRootViewController:viewController];
+                [self displayHomePage];
             }
         }];
     }
 }
 
+- (IBAction)continueWithFacebook:(UIButton *)sender {
+    FBSDKLoginManager *manager = [FBSDKLoginManager new];
+    FBSDKLoginConfiguration *configuration = [[FBSDKLoginConfiguration alloc] initWithPermissions:@[@"public_profile", @"email"] tracking:FBSDKLoginTrackingEnabled];
+
+    [self displayFacebookLoginWithManager:manager withConfiguration:configuration];
+}
+
+- (void)displayFacebookLoginWithManager:(FBSDKLoginManager *)manager
+                      withConfiguration:(FBSDKLoginConfiguration *)configuration {
+    [manager logInFromViewController:self
+                       configuration:configuration
+                          completion:^(FBSDKLoginManagerLoginResult *_Nullable result, NSError *_Nullable error) {
+        if (error == nil && !result.isCancelled) {
+            FBSDKGraphRequest *request = [[FBSDKGraphRequest alloc]
+                                          initWithGraphPath:@"/me"
+                                          parameters:@{@"fields":@"first_name,last_name,email"}
+                                          HTTPMethod:@"GET"];
+
+            [self fetchUserInformationWithRequest:request];
+        }
+    }];
+}
+
+- (void)fetchUserInformationWithRequest:(FBSDKGraphRequest *)request {
+    [request startWithCompletion:^(id<FBSDKGraphRequestConnecting> _Nullable connection, id _Nullable result, NSError *_Nullable error) {
+        if (result) {
+            self.requestResult = @{@"first_name":result[@"first_name"], @"last_name":result[@"last_name"], @"email":result[@"email"]};
+            NSString *email = result[@"email"];
+
+            [self handleFacebookLoginWithEmail:email];
+        }
+    }];
+}
+
+- (void)handleFacebookLoginWithEmail:(NSString *)email {
+    PFQuery *userQuery = [User query];
+
+    [userQuery whereKey:@"email" equalTo:email];
+
+    [userQuery countObjectsInBackgroundWithBlock:^(int count, NSError *_Nullable error) {
+        if (count == 0) {
+            [self performSegueWithIdentifier:@"signUpSegue" sender:self];
```

Comment:
This is not how I'd do it. If there isn't a user with that email then you should make one behind the scenes. You want there to be no difference to your end user regarding how they sign up. This makes them go through the signup even if they used fb login which is extra buttons for them to tap.